src/screens/Home/ResourcesItemScreen.js

```diff
@@ -0,0 +1,161 @@
+import React, { useState, useEffect, useCallback } from 'react';
+import {
+  View, StyleSheet, ScrollView, ActivityIndicator,
+} from 'react-native';
+import PropTypes from 'prop-types';
+import firestore from '@react-native-firebase/firestore';
+import {
+  Title, Subheading, Text, Paragraph, List, Caption,
+} from 'react-native-paper';
+import { theme } from '../../style';
+
+/* Resource item screen styles */
+const resourcesStyles = StyleSheet.create({
+  container: {
+    flex: 1,
+    backgroundColor: theme.colors.background,
+  },
+  itemContainer: {
+    flex: 1,
+    backgroundColor: 'white',
+    top: 400,
+    width: '100%',
+    borderRadius: 20,
+    padding: 20,
+    paddingBottom: 420,
+  },
+  authorName: {
+    textAlign: 'right',
+  },
+});
+
+/* Returns a single list item with appropriate icon */
+function ResourceStep({
+  stepNumber, title, description,
+}) {
+  return (
+    <List.Item
+      title={title}
+      description={description}
+      left={() => <List.Icon color={theme.colors.accent} icon={`numeric-${stepNumber}-circle`} />}
+    />
+  );
+}
+
+export default function ResourcesItemScreen(props) {
+  const { navigation } = props;
+  const resourceID = navigation.getParam('resourceID');
+
+  /* Resource item data */
+  const [title, setTitle] = useState(null);
+  const [description, setDescription] = useState(null);
+  const [type, setType] = useState(null);
+  const [body, setBody] = useState(null);
+  const [steps, setSteps] = useState(null);
+  const [subheader, setSubheader] = useState(null);
+  const [authorName, setAuthorName] = useState(null);
+
+  /* Screen state */
+  const [loading, setLoading] = useState(false);
+  const [errorMessage, setErrorMessage] = useState(null);
+
+  /* Query data from firestore */
+  const getData = useCallback(async () => {
+    setLoading(true);
+
+    try {
+      /* Query data from resource item collection */
+      const resourceSnap = await firestore().collection('resource_items').doc(resourceID).get();
+      const resourceData = resourceSnap.data();
+
+      /* Query user data */
+      const authorSnapshot = firestore().collection('users').doc(resourceData.userID).get();
+      const authorSnap = await Promise.all([authorSnapshot]);
+
+      setAuthorName(authorSnap.length ? authorSnap[0].get('displayName') : null);
+      setTitle(resourceData.title);
+      setDescription(resourceData.description);
+      setType(resourceData.type);
+      setSubheader(resourceData.subheader);
+
+      /* Different behavior if resource is of type STEPS or FREEFORM */
+      if (type === 'STEPS') {
+        const stepsList = await firestore().collection('resource_items').doc(resourceID).collection('steps')
```

Comment:
The ActivityIndicator isn't being shown while the steps are loading. Converting this to use .then() instead of await, and putting setLoading inside the .then would fix this.